alacritty_terminal/src/term/mod.rs

```diff
@@ -1743,6 +1762,147 @@ impl<T: EventListener> Handler for Term<T> {
     }
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum CollectorMode {
+    Sync,
+    Async(NonZeroUsize),
+}
+
+impl From<usize> for CollectorMode {
+    fn from(value: usize) -> Self {
+        match NonZeroUsize::new(value) {
+            Some(value) => Self::Async(value),
+            None => Self::Sync,
+        }
+    }
+}
+
+pub struct GridCollector {
```

Comment:
Why do we need a completely different structure for iterating over the grid and converting it to a string, instead of using what already exists?

This entire struct and its methods seem like a lot of complexity that is duplicated from other places and will easily haunt us in the future.